scm.go

```
@@ -888,16 +888,20 @@ func checkoutGitWithExecutor(
 		// TODO(pedge)
 		return fmt.Errorf("CouldNotClone: %v %v", err.Error(), cloneStderr.String())
 	}
+
+	// If we have a blank commitID then we want the current head of the branch so a checkout is not needed
 	var checkoutStderr bytes.Buffer
-	if err := executor.Execute(
-		&exec.Cmd{
-			Args:   []string{"git", "checkout", "-f", commitID},
-			SubDir: path,
-			Stderr: &checkoutStderr,
-		},
-	)(); err != nil {
-		// TODO(pedge)
-		return fmt.Errorf("CouldNotCheckout: %v %v", err.Error(), checkoutStderr.String())
+	if commitID != "" {
+		if err := executor.Execute(
+			&exec.Cmd{
+				Args:   []string{"git", "checkout", "-f", commitID},
+				SubDir: path,
+				Stderr: &checkoutStderr,
+			},
+		)(); err != nil {
+			// TODO(pedge)
```

I know this is outside the scope, but I wonder if we can tackle some of these `TODO(peter)`s. I guess the TODO is that he was just swallowing and logging all errors instead of returning them up the call stack?